src/velo-fed/VeloFarmerV3.sol

```diff
@@ -0,0 +1,715 @@
+        maxSlippageBpsUsdcNativeToDola = maxSlippageBps[2];
+        maxSlippageBpsDolaToUsdcNative = maxSlippageBps[3];
+        maxSlippageBpsUsdcToUsdcNative = maxSlippageBps[4];
+        maxSlippageBpsUsdcNativeToUsdc = maxSlippageBps[5];
+        maxSlippageBpsLiquidity = maxSlippageBpsLiquidity_;
+    }
+
+    modifier onlyGov() {
+        if (msg.sender != address(ovmL2CrossDomainMessenger) ||
+            ovmL2CrossDomainMessenger.xDomainMessageSender() != gov
+        ) revert OnlyGov();
+        _;
+    }
+
+    modifier onlyPendingGov() {
+        if (msg.sender != address(ovmL2CrossDomainMessenger) ||
+            ovmL2CrossDomainMessenger.xDomainMessageSender() != pendingGov
+        ) revert OnlyPendingGov();
+        _;
+    }
+
+    modifier onlyChair() {
+        if ((msg.sender != address(ovmL2CrossDomainMessenger) ||
+            ovmL2CrossDomainMessenger.xDomainMessageSender() != chair) &&
+            msg.sender != l2chair
+        ) revert OnlyChair();
+        _;
+    }
+
+    modifier onlyGovOrGuardian() {
+        if ((msg.sender != address(ovmL2CrossDomainMessenger) ||
+            (ovmL2CrossDomainMessenger.xDomainMessageSender() != gov) &&
+             ovmL2CrossDomainMessenger.xDomainMessageSender() != guardian)
+        ) revert OnlyGovOrGuardian();
+        _;
+    }
+
+    /**
+     * @notice Claims all VELO token rewards accrued by this contract & transfer all VELO owned by this contract to `treasury`
+     */
+    function claimVeloRewards() external {
+        dolaGauge.getReward(address(this));
+
+        IERC20(veloTokenAddr).transfer(treasury, IERC20(veloTokenAddr).balanceOf(address(this)));
+    }
+
+
+    /**
+     * @notice Attempts to deposit `dolaAmount` of DOLA & `usdcAmount` of USDC into Velodrome DOLA/USDC stable pool. Then, deposits LP tokens into gauge.
+     * @param dolaAmount Amount of DOLA to be added as liquidity in Velodrome DOLA/USDC pool
+     * @param usdcAmount Amount of USDC to be added as liquidity in Velodrome DOLA/USDC pool
+     */
+    function deposit(uint dolaAmount, uint usdcAmount) public onlyChair {
+        uint lpTokenPrice = getLpTokenPrice();
+
+        DOLA.approve(address(router), dolaAmount);
+        nUSDC.approve(address(router), usdcAmount);
+        (uint dolaSpent, uint usdcSpent, uint lpTokensReceived) = router.addLiquidity(address(DOLA), address(nUSDC), true, dolaAmount, usdcAmount, 0, 0, address(this), block.timestamp);
+        require(lpTokensReceived > 0, "No LP tokens received");
+
+        uint totalDolaValue = dolaSpent + (usdcSpent *DOLA_USDC_CONVERSION_MULTI);
+
+        uint expectedLpTokens = totalDolaValue *1e18 / lpTokenPrice *(PRECISION - maxSlippageBpsLiquidity) / PRECISION;
+        if (lpTokensReceived < expectedLpTokens) revert LiquiditySlippageTooHigh();
+        
+        uint lpBalance = LP_TOKEN_NATIVE.balanceOf(address(this));
+        LP_TOKEN_NATIVE.approve(address(dolaGauge), lpBalance);
+        dolaGauge.deposit(lpBalance);
+    }
```

Comment:
Consider making internal and adding `deposit(uint dolaAmount, uint usdcAmount) external onlyChair returns (uint);` function.